ts/private/ts_project.bzl

```diff
@@ -124,11 +124,25 @@ def _ts_project_impl(ctx):
         ])
         outputs.append(ctx.outputs.buildinfo_out)
     runtime_outputs = json_outs + js_outs + map_outs
-    typings_outputs = typings_outs + typing_maps_outs + [s for s in ctx.files.srcs if s.path.endswith(".d.ts")]
+    typings_srcs = [s for s in ctx.files.srcs if s.path.endswith(".d.ts")]
 
-    if len(js_outs) + len(typings_outputs) < 1:
+    if len(js_outs) + len(typings_outs) < 1:
         label = "//{}:{}".format(ctx.label.package, ctx.label.name)
-        if ctx.attr.transpile:
+        if len(typings_srcs) > 0:
+            no_outs_msg = """ts_project target {target} has no outputs, but it has typings in srcs.
```

Comment:
this makes me wonder whether tsc is supposed to type check .d.ts sources as well.

Reply:
I think it could - but it doesn't produce any outputs in that case, so it would have to be run as a Bazel "validation action" whose outputs aren't required in the dep graph. I'll note that in https://github.com/aspect-build/rules_ts/issues/85#issuecomment-1180520106

Reply:
Ah cool. I'd expect tsc to at least type check and copy it over if instructed. At least that is what I'd expect.